From da81268ec4ab9502844eac6fc9e51d945b149553 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 26 Nov 2014 00:53:39 -0500
Subject: [PATCH] Fix EAR AddToChunk bug

We should never skip a tick if the entity is flagged as "not added to chunk" so it can be added correctly.
---
 src/main/java/net/minecraft/server/Entity.java  | 2 +-
 src/main/java/org/spigotmc/ActivationRange.java | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index 153337b..fd07ca2 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -111,7 +111,7 @@ public abstract class Entity {
     protected DataWatcher datawatcher;
     private double g;
     private double h;
-    public boolean ag;
+    public boolean ag;public boolean isAddedToChunk() { return ag; } // Spigot
     public int ah;
     public int ai;
     public int aj;
diff --git a/src/main/java/org/spigotmc/ActivationRange.java b/src/main/java/org/spigotmc/ActivationRange.java
index f69cc65..40cc3b2 100644
--- a/src/main/java/org/spigotmc/ActivationRange.java
+++ b/src/main/java/org/spigotmc/ActivationRange.java
@@ -263,6 +263,10 @@ public class ActivationRange
     {
         if (entity.isDisabled) return true; // EMC
         SpigotTimings.checkIfActiveTimer.startTiming();
+        if (!entity.isAddedToChunk()) {
+            SpigotTimings.checkIfActiveTimer.stopTiming();
+            return true;
+        }
         boolean isActive = entity.activatedTick >= MinecraftServer.currentTick || entity.defaultActivationState;
 
         // Should this entity tick?
-- 
1.9.1

